drivers/netdev/ieee802154.c

```diff
@@ -284,4 +333,67 @@ int netdev_ieee802154_dst_filter(netdev_ieee802154_t *dev, const uint8_t *mhr)
     return 1;
 }
 
+#if IS_USED(MODULE_IEEE802154_SECURITY)
+/**
+ * @brief   Set the encryption key to be used for the next cipher operation
+ *
+ * @param[in] ctx       Security context
+ * @param[in] key       Key to be use for the next cipher operation
+ */
+static void _netdev_ieee802154_sec_set_key(ieee802154_sec_context_t *ctx,
+                                           const block16_t key)
+{
+    /* NOP */
+    (void)ctx;
+    (void)key;
+}
```

Comment:
This is now a callback implementation to be used in `ieee802154_radio_cipher_ops_t` For hardware support the key mus be written to the transceiver. For the software solution, there is nothing to do but, copying the key into the `cipher_t` struct. The copying is done in `static void _set_key()`, which wraps around the `set_key` callback.
Alternatively, I could set the callback to `NULL`, but then I must always check if it is indeed `NULL`, in `_set_key()`

Reply:
Makes sense. Since both @benpicco and I got puzzled by this, maybe it makes sense to add a comment. I mark this as resolved and leave @benpicco's instance open.